backend/index.js

```diff
@@ -5,13 +5,15 @@ const ratingRoutes = require("./routes/ratings");
 const usersRoutes = require("./routes/users");
 const appExpress = express();
 require("dotenv").config();
-appExpress.use(parser.json());
-appExpress.use("/rating", ratingRoutes);
-appExpress.use("/user", usersRoutes);
 
 // -------- define CORS for frontend connection -----------
 const cors = require("cors");
-appExpress.use(cors());
+appExpress.use(cors({ origin: process.env.FRONTEND_URL, credentials: true }));
+
+appExpress.use(parser.json());
+
+appExpress.use("/rating", ratingRoutes);
+appExpress.use("/user", usersRoutes);
```

Comment:
I moved `cors()` before the route definitions since cors need to be enabled first. Otherwise, the cors configuration isn't applied to the specified routes.

I also restricted the API access to the URL of the frontend (see `.env` file).